tsfresh/feature_extraction/feature_calculators.py

```diff
@@ -2212,3 +2213,59 @@ def benford_correlation(x):
     # np.corrcoef outputs the normalized covariance (correlation) between benford_distribution and data_distribution.
     # In this case returns a 2x2 matrix, the  [0, 1] and [1, 1] are the values between the two arrays
     return np.corrcoef(benford_distribution, data_distribution)[0, 1]
+
+
+@set_property("fctype", "combiner")
+def query_similarity_count(x, param):
+    """
+    This feature calculator accepts an input query subsequence parameter,
+    compares the query (under z-normalized Euclidean distance) to all
+    subsequences within the time series, and returns a count of the number
+    times the query was found in the time series (within some predefined
+    maximum distance threshold).
+
+    :param x: the time series to calculate the feature of
+    :type x: numpy.ndarray
+    :param param: contains dictionaries
+                  {"query": Q, "threshold": thr, "normalize": norm}
+                  with `Q` (numpy.ndarray), the query subsequence to compare the
+                  time series against. If `Q` is omitted then a value of zero
+                  is returned. Additionally, `thr` (float), the maximum
+                  z-normalized Euclidean distance threshold for which to
+                  increment the query similarity count. If `thr` is omitted
+                  then a default threshold of `thr=0.0` is used, which
+                  corresponds to finding exact matches to `Q`. Finally, for
+                  non-normalized (i.e., without z-normalization) Euclidean set
+                  `norm` (bool) to `False.
+    :type param: list
+    :return x: the different feature values
+    :return type: int
+    """
+    res = [0] * len(param)
+
+    for i, d in enumerate(param):
+        normalize = d.get("normalize", True)
+        if not isinstance(normalize, bool):
```

Comment:
I would be very careful with that. Python users are used to use "untypical types", as Python has a very sophisticated bool-cast system. Example: 0 and 1 are very common alternatives to False and True and work exactly the same way as the boolean constants. If you want to check for `None`, just use `if normalize is None` (or maybe I misunderstood your intention).

Reply:
Well, I'm allowing the user to omit the `normalize` parameter (i.e., `normalize` is optional and defaults to `True` when not provided). Currently, we allow three optional parameters: `query`, `threshold`, and `normalize` and they are all optional. It isn't clear to me if defaults are automatically taken from `settings.py` if some or all are missing or when a user provides a mixture of parameters in a list.

Reply:
I think this is fixed now.